src/middlewared/middlewared/plugins/ports/ports.py

```diff
@@ -40,20 +52,51 @@ async def get_in_use(self):
 
         return ports + self.SYSTEM_USED_PORTS
 
-    async def validate_port(self, schema, value, whitelist_namespace=None):
+    async def validate_port(self, schema, port, bindip='0.0.0.0', whitelist_namespace=None, raise_error=False):
         verrors = ValidationErrors()
-        for port_attachment in await self.middleware.call('port.get_in_use'):
-            if value in port_attachment['ports'] and port_attachment['namespace'] != whitelist_namespace:
-                for port_entry in filter(
-                    lambda p: value in p['ports'],
-                    port_attachment['port_details']
-                ):
-                    err = 'The port is being used by '
-                    if port_entry['description']:
-                        err += f'{port_entry["description"]!r} in {port_attachment["title"]!r}'
-                    else:
-                        err += f'{port_attachment["title"]!r}'
-
-                    verrors.add(schema, err)
+        bindip_protocol = get_ip_protocol(bindip)
+        wildcard_ip = '0.0.0.0' if bindip_protocol == 4 else '::'
+        port_mapping = await self.ports_mapping(whitelist_namespace)
+        port_attachment = port_mapping[port]
+        if not any(
+            get_ip_protocol(ip) == bindip_protocol for ip in port_attachment
+        ) or (
+            bindip not in port_attachment and wildcard_ip not in port_attachment and bindip != wildcard_ip
+        ):
+            return verrors
+
+        ip_errors = []
+        for index, port_detail in enumerate(port_attachment.items()):
+            ip, port_entry = port_detail
+            if get_ip_protocol(ip) != bindip_protocol:
+                continue
+
+            if bindip == wildcard_ip or ip == wildcard_ip or (bindip != wildcard_ip and ip == bindip):
+                entry = next(
```

Comment:
This will raise a `StopIteration` if a match isn't found. Are you expecting this method to crash here with that type of traceback?